File: J7.c
#include <stdio.h>
#include<math.h>
int main()
{
int t,i,P,S;
double V,C;
scanf("%d",&t);
for(i=0;i<t;i++)
{
scanf("%d %d",&P,&S);
C=(double)(P-sqrt(P*P-24*S))/12;
V=C*C*((double)(P/4)-2*C);
printf("%.2lf\n",V);
}
return 0;
}

File: j7.py
import math
t=int(input())
while t>0:
    p,s=input().split(" ")
    p=int(p)
    s=int(s)
    c=(p-math.sqrt(p*p-24*s))/12
    v=c*c*(p/4-2*c)
    if(v-math.floor(v)>=0.005):
        print(math.ceil(v*100)/100)
    else:
        print(math.floor(v*100)/100)
    t-=1
